Content.Client/SS220/SpaceWars/Party/UI/PartyUIController.cs

```diff
@@ -18,37 +19,94 @@ namespace Content.Client.SS220.SpaceWars.Party.UI;
 {
     [Dependency] private readonly IPartyManager _party = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
 
     private PartyWindow? _window;
     private MenuButton? GamePartyButton => UIManager.GetActiveUIWidgetOrNull<GameTopMenuBar>()?.PartyButton;
     private Button? LobbyPartyButton => (UIManager.ActiveScreen as LobbyGui)?.PartyButton;
 
-    private bool _bindsRegistered;
-    private bool _hasUnreadInfo;
-
     public static readonly Color DefaultBackgroundColor = new(60, 60, 60);
     public static readonly Color DefaultInnerBackgroundColor = new(32, 32, 40);
 
+    private static readonly TimeSpan ClosedWindowLifeTime = TimeSpan.FromSeconds(15);
+    private TimeSpan _windowDisposeTime = TimeSpan.Zero;
+
+    private bool _bindsRegistered;
+    private bool _hasUnreadInfo;
+
     public override void Initialize()
     {
         base.Initialize();
 
         _party.LocalPartyChanged += OnLocalPartyChanged;
-        _party.ChatMessageReceived += _ => UnreadInfoReceived();
 
-        _party.ReceivedInviteAdded += _ => UnreadInfoReceived();
-        _party.ReceivedInviteRemoved += _ => UnreadInfoReceived();
+        _party.LocalPartyEvents.HostChanged += _ => UnreadInfoReceived();
+
+        _party.LocalPartyEvents.MemberAdded += _ => UnreadInfoReceived();
+        _party.LocalPartyEvents.MemberRemoved += _ => UnreadInfoReceived();
+
+        _party.LocalPartyEvents.InviteAdded += _ => _window?.MainTab.LocalPartyInvitesWindow?.Refresh();
+        _party.LocalPartyEvents.InviteRemoved += _ => _window?.MainTab.LocalPartyInvitesWindow?.Refresh();
+        _party.LocalPartyEvents.InviteUpdated += OnLocalPartyInviteUpdated;
+
+        _party.ReceivedInviteAdded += OnReceivedInviteAdded;
+        _party.ReceivedInviteRemoved += OnReceivedInviteRemoved;
+        _party.ReceivedInviteUpdated += OnReceivedInviteUpdated;
+
+        _party.ChatMessageReceived += OnChatMessageReceived;
     }
 
-    private void OnLocalPartyChanged(Party? party)
+    public override void FrameUpdate(FrameEventArgs args)
     {
-        if (party != null)
-        {
-            party.HostChanged += _ => UnreadInfoReceived();
-            party.MemberAdded += _ => UnreadInfoReceived();
-            party.MemberRemoved += _ => UnreadInfoReceived();
-        }
+        base.FrameUpdate(args);
+
+        if (_window != null && !_window.IsOpen && _gameTiming.CurTime >= _windowDisposeTime)
+            _window = null;
+    }
+
+    private void OnLocalPartyChanged((Party? Old, Party? New) tulpe)
+    {
+        UnreadInfoReceived();
+        _window?.Refresh();
+    }
+
+    private void OnLocalPartyInviteUpdated(PartyInvite invite)
+    {
+        if (_window?.MainTab.LocalPartyInvitesWindow is not { } localPartyInvitesWindow)
+            return;
+
+        if (localPartyInvitesWindow.TryGetEntry(invite, out var entry) is true)
```

Comment:
```suggestion
        if (localPartyInvitesWindow.TryGetEntry(invite, out var entry))
```
Кто украл кнопку suggestion я тебе руками всё пишу